File: scripts/run_all.py
#!/usr/bin/env python3
"""
run_all.py - unified loader for the awesome-regulatory-monitors actor fleet.

Runs one or more actors from the directory in README.md with a shared
onlyNew=true delta flag, and prints a consolidated summary across sources.
Each actor's own real input fields are respected - this script only sets
the fields that are common across the fleet (maxItems-style caps and the
delta flag); anything actor-specific should be passed via --extra-input.

Usage:
    pip install apify-client
    export APIFY_TOKEN=your_token_here
    python run_all.py --actors actor-19-maritime-sanctions-monitor actor-22-drug-safety-recalls-monitor
    python run_all.py --all
"""

import argparse
import json
import os
import sys

from apify_client import ApifyClient

# Real actor directory, actor slug -> its real delta-flag field name (NOT
# all identical across this fleet - actor-18 uses skipKnownLeads instead of
# onlyNew, since it has no content-change concept, only a "seen before"
# one; primer-actor uses onlyChanged, since it tracks per-URL content
# change rather than a registry listing).
ACTOR_DELTA_FLAGS = {
    "australia-grantconnect-monitor": "onlyNew",
    "uk-hse-enforcement-monitor": "onlyNew",
    "florida-tenders-monitor": "onlyNew",
    "santafe-compras-monitor": "onlyNew",
    "tucuman-compras-monitor": "onlyNew",
    "salta-compras-monitor": "onlyNew",
    "mendoza-compras-monitor": "onlyNew",
    "entrerios-compras-monitor": "onlyNew",
    "cordoba-compras-monitor": "onlyNew",
    "pba-tenders-monitor": "onlyNew",
    "diario-oficial-cl-monitor": "onlyNew",
    "primer-actor": "onlyChanged",
    "actor-18-b2b-lead-magnet": "skipKnownLeads",
    "actor-19-maritime-sanctions-monitor": "onlyNew",
    "actor-20-mdb-procurement-monitor": "onlyNew",
    "actor-21-patent-ip-enforcement-monitor": "onlyNew",
    "actor-22-drug-safety-recalls-monitor": "onlyNew",
}

OWNER = "stefano_seggio"


def run_one(client: ApifyClient, actor_slug: str, extra_input: dict) -> dict:
    delta_flag = ACTOR_DELTA_FLAGS.get(actor_slug)
    run_input = dict(extra_input)
    if delta_flag and delta_flag not in run_input:
        run_input[delta_flag] = True

    print(f"--- Running {actor_slug} (input: {json.dumps(run_input)}) ---")
    run = client.actor(f"{OWNER}/{actor_slug}").call(run_input=run_input)
    items = list(client.dataset(run["defaultDatasetId"]).iterate_items())
    print(f"    {len(items)} item(s) returned, run status: {run['status']}")
    return {"actor": actor_slug, "run_id": run["id"], "item_count": len(items), "status": run["status"]}


def main():
    parser = argparse.ArgumentParser(description="Run one or more actors from the awesome-regulatory-monitors fleet.")
    parser.add_argument("--actors", nargs="*", default=[], help="Specific actor slugs to run (see README.md for the full list)")
    parser.add_argument("--all", action="store_true", help="Run every actor in the directory (real cost - each run bills per this actor's own pricing)")
    parser.add_argument("--extra-input", type=str, default="{}", help="JSON object of additional input fields applied to every run, e.g. '{\"maxItems\": 50}'")
    args = parser.parse_args()

    token = os.environ.get("APIFY_TOKEN")
    if not token:
        print("Error: set the APIFY_TOKEN environment variable first.", file=sys.stderr)
        sys.exit(1)

    actors_to_run = list(ACTOR_DELTA_FLAGS.keys()) if args.all else args.actors
    if not actors_to_run:
        print("Nothing to run - pass --actors <slug...> or --all. See README.md for the full actor directory.", file=sys.stderr)
        sys.exit(1)

    extra_input = json.loads(args.extra_input)
    client = ApifyClient(token)

    results = [run_one(client, slug, extra_input) for slug in actors_to_run]

    print("\n=== Summary ===")
    for r in results:
        print(f"{r['actor']}: {r['item_count']} item(s), status={r['status']}")


if __name__ == "__main__":
    main()

File: scripts/runAll.js
#!/usr/bin/env node
/**
 * runAll.js - unified loader for the awesome-regulatory-monitors actor fleet.
 *
 * Runs one or more actors from the directory in README.md with a shared
 * delta flag, and prints a consolidated summary across sources. Each
 * actor's own real input fields are respected - this script only sets the
 * delta flag and whatever common fields you pass via --extra-input.
 *
 * Usage:
 *   npm install apify-client
 *   export APIFY_TOKEN=your_token_here
 *   node runAll.js --actors actor-19-maritime-sanctions-monitor actor-22-drug-safety-recalls-monitor
 *   node runAll.js --all
 */

import { ApifyClient } from 'apify-client';

const OWNER = 'stefano_seggio';

// Real actor directory, actor slug -> its real delta-flag field name (NOT
// identical across the fleet - actor-18 uses skipKnownLeads (no
// content-change concept, only "seen before"); primer-actor uses
// onlyChanged (per-URL content-change tracking, not a registry listing).
const ACTOR_DELTA_FLAGS = {
    'australia-grantconnect-monitor': 'onlyNew',
    'uk-hse-enforcement-monitor': 'onlyNew',
    'florida-tenders-monitor': 'onlyNew',
    'santafe-compras-monitor': 'onlyNew',
    'tucuman-compras-monitor': 'onlyNew',
    'salta-compras-monitor': 'onlyNew',
    'mendoza-compras-monitor': 'onlyNew',
    'entrerios-compras-monitor': 'onlyNew',
    'cordoba-compras-monitor': 'onlyNew',
    'pba-tenders-monitor': 'onlyNew',
    'diario-oficial-cl-monitor': 'onlyNew',
    'primer-actor': 'onlyChanged',
    'actor-18-b2b-lead-magnet': 'skipKnownLeads',
    'actor-19-maritime-sanctions-monitor': 'onlyNew',
    'actor-20-mdb-procurement-monitor': 'onlyNew',
    'actor-21-patent-ip-enforcement-monitor': 'onlyNew',
    'actor-22-drug-safety-recalls-monitor': 'onlyNew',
};

function parseArgs(argv) {
    const args = { actors: [], all: false, extraInput: {} };
    for (let i = 0; i < argv.length; i++) {
        if (argv[i] === '--all') args.all = true;
        else if (argv[i] === '--actors') {
            while (argv[i + 1] && !argv[i + 1].startsWith('--')) args.actors.push(argv[++i]);
        } else if (argv[i] === '--extra-input') {
            args.extraInput = JSON.parse(argv[++i]);
        }
    }
    return args;
}

async function runOne(client, actorSlug, extraInput) {
    const deltaFlag = ACTOR_DELTA_FLAGS[actorSlug];
    const runInput = { ...extraInput };
    if (deltaFlag && !(deltaFlag in runInput)) runInput[deltaFlag] = true;

    console.log(`--- Running ${actorSlug} (input: ${JSON.stringify(runInput)}) ---`);
    const run = await client.actor(`${OWNER}/${actorSlug}`).call(runInput);
    const { items } = await client.dataset(run.defaultDatasetId).listItems();
    console.log(`    ${items.length} item(s) returned, run status: ${run.status}`);
    return { actor: actorSlug, runId: run.id, itemCount: items.length, status: run.status };
}

async function main() {
    const args = parseArgs(process.argv.slice(2));
    const token = process.env.APIFY_TOKEN;
    if (!token) {
        console.error('Error: set the APIFY_TOKEN environment variable first.');
        process.exit(1);
    }

    const actorsToRun = args.all ? Object.keys(ACTOR_DELTA_FLAGS) : args.actors;
    if (actorsToRun.length === 0) {
        console.error('Nothing to run - pass --actors <slug...> or --all. See README.md for the full actor directory.');
        process.exit(1);
    }

    const client = new ApifyClient({ token });
    const results = [];
    for (const slug of actorsToRun) {
        results.push(await runOne(client, slug, args.extraInput));
    }

    console.log('\n=== Summary ===');
    for (const r of results) {
        console.log(`${r.actor}: ${r.itemCount} item(s), status=${r.status}`);
    }
}

main().catch((error) => {
    console.error(error);
    process.exit(1);
});
